### PR TITLE
(NETDEV-30) Enhance syslog, tacacs, radius

### PR DESCRIPTION
**Pull request by shermdog**
This PR enhances the existing syslog, tacacs, and radius types as specified in  puppetlabs/netdev_stdlib#25

syslog_server
- port

syslog_settings
- console
- monitor
- source_interface
- vrf

tacacs_global
- source_interface

radius_global
- source_interface

### CODE REVIEW

**Comment by mikewiebe** [comment]
@shermdog Sorry for the delay on the review.  We are up against a release for a different project and have not been able to carve out time but I will review this on Monday.

File: tests/beaker_tests/radius_global/radius_global_provider_defaults.rb [hunk 1]
```diff
@@ -58,55 +58,23 @@
...
-
-  # @step [Step] Requests manifest from the master server to the agent.
-  step 'TestStep :: Get resource present (with changes)manifest from master' do
-    # Expected exit_code is 0 since this is a bash shell cmd.
-    on(master, RadiusGlobalLib.create_radius_global_manifest_change)
-
-    # Expected exit_code is 2 since this is a puppet agent cmd with change.
-    cmd_str = PUPPET_BINPATH + 'agent -t'
-    on(agent, cmd_str, acceptable_exit_codes: [2])
+  # To remove a configured key we have ot know the key value
```
Comment (mikewiebe) [inline tests/beaker_tests/radius_global/radius_global_provider_defaults.rb]: `/ot/to/`

File: tests/beaker_tests/radius_global/radius_global_provider_nondefaults.rb [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+testheader = 'radius_global Resource :: All Attributes Defaults'
+
+def cleanup
+  logger.info('Testcase Cleanup:')
+
+  command_config(agent, 'radius-server timeout 5')
+  command_config(agent, 'radius-server retransmit 1')
+  command_config(agent, 'no ip radius source-interface')
+
+  # To remove a configured key we have ot know the key value
```
Comment (mikewiebe) [inline tests/beaker_tests/radius_global/radius_global_provider_nondefaults.rb]: `/ot/to/`

**Comment by mikewiebe** [comment]
@shermdog Just a few very minor nits, otherwise 👍

**Comment by mikewiebe** [comment]
For future PR, I would prefer we write new tests or refactor existing tests to use the new style beaker tests.  Here is an example:

https://github.com/cisco/cisco-network-puppet-module/blob/develop/tests/beaker_tests/cisco_bgp_neighbor/test_bgpneighbor.rb

**Comment by mikewiebe** [comment]
@shermdog Can you also rev the netdev version in the metadata.json file as part of these diffs?

https://github.com/cisco/cisco-network-puppet-module/blob/develop/metadata.json#L11

**Comment by shermdog** [comment]
@mikewiebe metadata bumped to released netdev_stdlib module (0.13.0) and CHANGELOG conflict resolved.

**Comment by mikewiebe** [comment]
👍

**Pull request merged**